src/prefect/environments/storage/local.py

```diff
@@ -31,14 +31,21 @@ class Local(Storage):
             absolute path and created.  Defaults to `True`
         - secrets (List[str], optional): a list of Prefect Secrets which will be used to populate `prefect.context`
             for each flow run.  Used primarily for providing authentication credentials.
+        - add_hostname_label (bool): if `True`, include the current host's hostname
```

Comment:
Instead of specifying the presence/absence of a specific label, we might add a general flag for whether to add default labels? This kwarg name could be used in other places in prefect where we might want to disable the automatic labels as well, providing some uniformity. Perhaps `add_default_labels=True` or `no_default_labels=False`?

Reply:
Good w/ me! Haven't taken a look at any other default labels, but I'm liking your idea more. We could create a new property on the base `Storage` class that could look like the following:

```python
class Storage:
     def __init__(self, result: Result = None, secrets: List[str] = None, labels: List[str] = None # new) -> None:
         self.result = result
         self.secrets = secrets or []
         # New
         self._labels = labels or []

    # New (changed)
    @property
    def labels(self) -> List[str]:
         # No longer would be abstract, and move that dependency to the `default_labels` method?
        return set(default_labels).update(set(self._labels))

    @property
    def default_labels(self):
        raise NotImplementedError()
```

We could then handle the union of the 2 list of labels (I'm assuming order of the labels doesn't matter here, but I don't know that for sure). 

Thoughts?

Reply:
Currently storage objects don't support setting labels on them at all (environments do) - the only labels ever on a storage class are autogenerated by the class. We could add a `labels` kwarg to all the `Storage` objects for users to add their own labels as you've done above, but since a flow run is environment + storage, I'm not sure if that's needed? No strong thoughts.

Handling the presence/absence of default labels for storage in the base class sounds fine though.

Reply:
One thing to mention too is that it appears the `Agents` don't separate the concept of a `Storage` label vs an `Environment` label. In the `LocalAgent` code, the label list gets extended with the 3 cloud storage magic strings (`azure-flow-storage`, `gcs-flow-storage`, `s3-flow-storage`) as well when checking for work. Although the actual check of whether the agent can deploy the flow is done by loading the flow's `Storage`, it seems like the only part that _could_ matter there is if people are also labeling their `Environment`'s with any of those magic strings? Probably a slim chance though.

I also don't have any strong opinions on it, but I also don't leverage the full functionality behind Prefect, so I'm not sure what the implications would be of tying these sets of labels together. I can put together the base implementation under the assumption that the users can define their label list twice and we can have a more targeted discussion after seeing that implementation?

Reply:
> I can put together the base implementation under the assumption that the users can define their label list twice and we can have a more targeted discussion after seeing that implementation?

Makes sense to me. Thanks for working on this.